packages/plugin-credentials/test/integration/spec/credentials.js

```diff
@@ -6,21 +6,71 @@
 
 import '../..';
 
-import CiscoSpark from '@ciscospark/spark-core';
 import {assert} from '@ciscospark/test-helper-chai';
+import testUsers from '@ciscospark/test-helper-test-users';
+import CiscoSpark from '@ciscospark/spark-core';
+import {apiScope} from '../..';
 
 describe(`plugin-credentials`, () => {
   describe(`Credentials`, () => {
+    describe(`#refresh()`, () => {
+      let spark;
+
+      beforeEach(() => testUsers.create({count: 1})
+        .then((users) => {
+          spark = new CiscoSpark({
+            credentials: {
+              supertoken: users[0].token
+            }
+          });
+        }));
+
+      let apiToken, kmsToken, supertoken;
+      beforeEach(() => {
+        supertoken = spark.credentials.supertoken.access_token;
+        apiToken = spark.credentials.userTokens.get(apiScope);
+        apiToken = apiToken && apiToken.access_token;
+        kmsToken = spark.credentials.userTokens.get(`spark:kms`);
+        kmsToken = kmsToken && kmsToken.access_token;
+      });
+
+      it(`refreshes the supertoken and all child tokens`, () => spark.credentials.refresh()
+        .then(() => {
+          assert.notEqual(spark.credentials.supertoken.access_token, supertoken);
+          assert.isDefined(spark.credentials.userTokens.get(apiScope));
+          assert.notEqual(spark.credentials.userTokens.get(apiScope).access_token, apiToken);
+          assert.isDefined(spark.credentials.userTokens.get(apiScope));
+          assert.isDefined(spark.credentials.userTokens.get(`spark:kms`));
+          assert.notEqual(spark.credentials.userTokens.get(`spark:kms`).access_token, supertoken);
+          assert.isDefined(spark.credentials.userTokens.get(`spark:kms`));
```

Comment:
the isDefined is for debugging; getting a `cannot read access token of undefined` error repeatedly made it harder to figure out where exactly i had made my mistake but the equality check is required to make sure the code actually does what was intended.